core/arch/arm/kernel/secure_partition.c

```diff
@@ -68,6 +85,32 @@ struct sp_session *sp_get_session(uint32_t session_id)
 	return NULL;
 }
 
+TEE_Result sp_get_partitions_info(struct ffa_partition_info *fpi,
+				  size_t *elements)
+{
+	size_t in_elements = *elements;
+	struct sp_session *s = NULL;
+	uint32_t count = 0;
+
+	TAILQ_FOREACH(s, &open_sp_sessions, link) {
+		if (count < in_elements) {
+			if (s->state != sp_dead) {
+				spmc_fill_partition_entry(fpi, s->endpoint_id);
+				fpi++;
+			}
+		}
+		if (s->state != sp_dead)
+			count++;
+	}
+
+	if (count > in_elements) {
+		*elements = count;
```

Comment:
The spec requires this to be returned on success too.

Reply:
Done